src/equations/kdv_1d.jl

```diff
@@ -119,6 +119,139 @@ function source_terms_manufactured(q, x, t, equations::KdVEquation1D)
     return SVector(s1)
 end
 
+"""
+    nondim2prim(u, equations::KdVEquation1D)
+
+Convert the non-dimensional variable `u` to the primitive/physical variable `eta`
+(total water height) for the [`KdVEquation1D`](@ref).
+
+The transformation is given by:
+```math
+\\eta = D(u - \\frac{2}{3})
+```
+where `D` is the still-water depth.
+
+!!! warning "Parameter constraints"
+    This conversion is only valid for equations with specific parameter values:
+    - `gravity = 4/27` 
+    - `D = 3.0` 
+    
+    These values ensure the dimensional KdV equation matches the standard 
+    non-dimensional form `u_t + u u_x + u_{xxx} = 0`.
+
+This function allows converting solutions from the standard non-dimensional 
+KdV form commonly found in literature to the dimensional form implemented 
+in DispersiveShallowWater.jl.
+
+See also [`prim2nondim`](@ref).
+"""
+function nondim2prim(u, equations::KdVEquation1D)
+    eta = @. equations.D * (u - 2 / 3)
+    return eta
+end
+
+"""
+    prim2nondim(eta, equations::KdVEquation1D)
+
+Convert the primitive/physical variable `eta` (total water height) to the 
+non-dimensional variable `u` for the [`KdVEquation1D`](@ref).
+
+The transformation is given by:
+```math
+u = \\frac{\\eta}{D} + \\frac{2}{3}
+```
+where `D` is the still-water depth.
+
+!!! warning "Parameter constraints"
+    This conversion is only valid for equations with specific parameter values:
+    - `gravity = 4/27` 
+    - `D = 3.0`
+    
+    These values ensure the dimensional KdV equation matches the standard 
+    non-dimensional form `u_t + u u_x + u_{xxx} = 0`.
+
+This function allows converting solutions from the dimensional form implemented 
+in DispersiveShallowWater.jl to the standard non-dimensional KdV form 
+commonly found in literature, enabling comparison with theoretical results 
+and other implementations.
+
+See also [`nondim2prim`](@ref).
+"""
+function prim2nondim(eta, equations::KdVEquation1D)
+    u = @. eta / equations.D + 2 / 3
+    return u
+end
+
+"""
+    prim2nondim(sol::ODESolution, equations::KdVEquation1D)
```

Comment:
As you said, I would also vote for removing this function. We don't have such a functions for the other conversions like `prim2cons` or `prim2phys`. So I would also not define it for `prim2nondim`.